sdks/java/io/hadoop-output-format/src/test/java/org/apache/beam/sdk/io/hadoop/outputformat/HadoopOutputFormatIOTest.java

```diff
@@ -0,0 +1,202 @@
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.mapreduce.OutputFormat;
+import org.apache.hadoop.mapreduce.lib.output.TextOutputFormat;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Unit tests for {@link HadoopOutputFormatIO}. */
+@RunWith(JUnit4.class)
+public class HadoopOutputFormatIOTest {
+  private static SerializableConfiguration serConf;
+
+  @Rule public final transient TestPipeline p = TestPipeline.create();
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  @BeforeClass
+  public static void setUp() {
+    EmployeeOutputFormat.initWrittenOutput();
+    serConf = loadTestConfiguration(EmployeeOutputFormat.class, Text.class, Employee.class);
+  }
+
+  private static SerializableConfiguration loadTestConfiguration(
+      Class<?> outputFormatClassName, Class<?> keyClass, Class<?> valueClass) {
+    Configuration conf = new Configuration();
+    conf.setClass("mapreduce.job.outputformat.class", outputFormatClassName, OutputFormat.class);
+    conf.setClass("mapreduce.job.outputformat.key.class", keyClass, Object.class);
+    conf.setClass("mapreduce.job.outputformat.value.class", valueClass, Object.class);
+    return new SerializableConfiguration(conf);
+  }
+
+  @Test
+  public void testWriteBuildsCorrectly() {
+    HadoopOutputFormatIO.Write<Text, Employee> write =
+        HadoopOutputFormatIO.<Text, Employee>write().withConfiguration(serConf.get());
+
+    assertEquals(serConf.get(), write.getConfiguration().get());
+    assertEquals(EmployeeOutputFormat.class, write.getOutputFormatClass().getRawType());
+    assertEquals(Text.class, write.getOutputFormatKeyClass().getRawType());
+    assertEquals(Employee.class, write.getOutputFormatValueClass().getRawType());
+  }
+
+  /**
+   * This test validates {@link HadoopOutputFormatIO.Write Write} transform object creation fails
+   * with null configuration. {@link HadoopOutputFormatIO.Write#withConfiguration(Configuration)
+   * withConfiguration(Configuration)} method checks configuration is null and throws exception if
+   * it is null.
+   */
+  @Test
+  public void testWriteObjectCreationFailsIfConfigurationIsNull() {
+    thrown.expect(IllegalArgumentException.class);
+    thrown.expectMessage("Configuration can not be null");
+    HadoopOutputFormatIO.<Text, Employee>write().withConfiguration(null);
+  }
+
+  /**
+   * This test validates functionality of {@link
+   * HadoopOutputFormatIO.Write#withConfiguration(Configuration) withConfiguration(Configuration)}
+   * function when Hadoop OutputFormat class is not provided by the user in configuration.
+   */
+  @Test
+  public void testWriteValidationFailsMissingInputFormatInConf() {
```

Comment:
Is this inputFormat or OutputFormat (jdoc / method name incoherence)? If it is inputFormat then why is it not tested in validate method using `checkArgument`?

Reply:
Thanks, it's a typo, fixed.